iterator.js

```diff
@@ -19,6 +19,10 @@ function Iterator(iterable) {
 
     if (iterable instanceof Iterator) {
         return iterable;
+    } else if (iterable.reduce) {
+        this.reduce = function() {
+            return iterable.reduce.apply(iterable,arguments);
+        }
```

Comment:
What is the intended effect of adding a reduce method to an iterator if the given iterable implements reduce?